### PR TITLE
Mm 43597

### PR DESCRIPTION
**Pull request by neallred**
#### Summary
For Cloud instances that hit the message history limit, show the limits reached modal to admins that are doing a fresh login. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-43597

#### Screenshots
Clip of admin not seeing the modal until they re log-in, and of regular user (nn2) not seeing the modal at all on log in:

Clip of admin not seeing the modal until they re log-in, and of regular user (nn2) not seeing the modal at all on log in:

#### Release Note
```release-note
For Cloud instances that hit the message history limit, show the limits reached modal to admins that are doing a fresh login. 
```

### CODE REVIEW

File: components/root/root.jsx [hunk 1]
```diff
@@ -542,6 +543,7 @@ export default class Root extends React.PureComponent {
                         )}
                         <ModalController/>
                         <GlobalHeader/>
+                        <CloudEffects/>
```
Comment (neallred) [inline components/root/root.jsx:546]: These need to run regardless of the product the user is in, so placed them here. Made CloudEffects itself as lightweight as possible (only checks whether user is logged in and the instance is cloud) to not impact performance for non-cloud and non-admin users.

**Review by AGMETEOR — APPROVED** [review]
Code LGTM

**Review by stevemudie — APPROVED** [review]
Looks good @neallred!

Tested and passed following the steps laid out in the ticket.

New test case [MM-T5105](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T5105) in place in Zephyr Scale.

File: components/cloud_effects/index.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+export default function CloudEffectsWrapper() {
+    const isCloud = useSelector(isCurrentLicenseCloud);
+    const currentUser = useSelector(getCurrentUser);
+
+    if (!isCloud || !currentUser) {
+        return null;
+    }
+
+    if (!isSystemAdmin(currentUser.roles)) {
```
Comment (nickmisasi) [inline components/cloud_effects/index.tsx:21]: nit: there's a shorthand selector for `isCurrentUserSystemAdmin` you could use here

**Review by nickmisasi — APPROVED** [review]
LGTM!

**Pull request merged**